swift_scality_backend/diskfile.py

```diff
@@ -178,7 +178,8 @@ def unexpected_http_status(response):
             conn = swift.common.bufferedhttp.http_connect_raw(
                 address, port, method, safe_path, headers)
 
-        with contextlib.closing(conn), eventlet.Timeout(self.proxy_timeout):
+        with contextlib.nested(contextlib.closing(conn),
```

Comment:
Hmh... To tackle this, and get rid of the warning, maybe rewrite it slightly into
```python
with contextlib.closing(conn):
    with eventlet.Timeout(self.proxy_timeout):
        resp = conn.getresponse()
        ...
```